map_test.go

```diff
@@ -830,6 +831,11 @@ func TestMapPinning(t *testing.T) {
 		t.Fatal("Can't write value:", err)
 	}
 
+	// This is a terrible hack: if loading a pinned map tries to load BTF,
+	// it will get a nil *btf.Spec from this *btf.Map. This is turn will make
+	// btf.NewHandle fail.
+	spec.BTF = new(btf.Map)
+
```

Comment:
Can you instead provide `btf.Spec` and check map info after opening the pinned map to confirm that it was not loaded?

Reply:
Unfortunately not, let me try to explain with pseudo code. Right now, the library does this:

```
btf := loadBTFForMap()
if mapIsPinned()
    return loadPinnedMap() // this doesn't use btf at all
}
return createMap(btf)
```

Afterwards:

```
if mapIsPinned()
    return loadPinnedMap()
}
btf := loadBTFForMap()
return createMap(btf)
```

Reply:
Okay, I see, so you are trying to assert that BTF is not loaded by providing an invalid BTF object.